### ISSUE TITLE
Extra or misplaced structures in specification files are not raised as warnings

### ISSUE THREAD

**Issue by dgreco** [guardrail-dev/guardrail#81]
I tried to generate the code using the server2.yaml swagger file under ./modules/sample/src/test/resources looking at the code I was expecting to have an handler able to return a sequence of Address, but the get addresses method has been generated for returning only 200 OK and not a sequence of Address. Is it an issue? Am I missing anything?

**Comment by stanislav-chetvertkov** [comment]
@dgreco 
This looks like an incorrectly defined schema to me

in the yaml file I changed
``` 
 /address:
    get:
      operationId: getAddresses
      x-scala-package: addresses
      responses:
        '200':
          type: array
          items:
            $ref: '#/definitions/Address'
        '404':
          description: Not found
```

to
```
  /address:
    get:
      operationId: getAddresses
      x-scala-package: addresses
      responses:
        '200':
          schema:
            type: array
            items:
              $ref: "#/definitions/Address"
        '404':
          description: Not found
```

And the signature in AddressesClient now looks correct

```
  def getAddresses(traceBuilder: TraceBuilder,
                   methodName: String = "get-addresses",
                   headers: scala.collection.immutable.Seq[HttpHeader] = Nil): EitherT[Future, Either[Throwable, HttpResponse], IndexedSeq[Address]]
```

**Comment by dgreco** [comment]
Ah ok thanks!
By the way I took that schema from the guardrail source distribution and the parser is not complaining ...

David Greco
d.greco@icloud.com
greco@acm.org
d.greco@computer.org
david.greco@postecert.it
https://github.com/dgreco
http://www.linkedin.com/in/davgreco
Twitter: davidgreco

**Comment by blast-hardcheese** [comment]
@dgreco Yeah -- we're currently using the official Swagger 2.0 parser, but unfortunately, that parser doesn't ensure totality. If access to the unconsumed JSON data were to be made available, we'd be able to provide much better error reporting.

**Comment by blast-hardcheese** [comment]
After investigating a bit, it seems as though the new parser [registers extra keys](https://github.com/swagger-api/swagger-parser/blob/2.0/modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/util/OpenAPIDeserializer.java#L192-L194). This will be good once #80 is resolved, as it seems the current version of the parser doesn't represent this.
